http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/conn/ClientConnectionRequestFactory.java

```diff
@@ -69,22 +80,27 @@ static ConnectionRequest wrap(ConnectionRequest orig) {
         @Override
         public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
             try {
-                // TODO v2 service metrics
-                //                if ("get".equals(method.getName())) {
-                //                    ServiceLatencyProvider latencyProvider = new ServiceLatencyProvider(
-                //                            AWSServiceMetrics.HttpClientGetConnectionTime);
-                //                    try {
-                //                        return method.invoke(orig, args);
-                //                    } finally {
-                //                        AwsSdkMetrics.getServiceMetricCollector()
-                //                                .collectLatency(latencyProvider.endTiming());
-                //                    }
-                //                }
-                return method.invoke(orig, args);
+                Instant startGet = null;
+                if ("get".equals(method.getName())) {
+                    startGet = Instant.now();
+                }
+                try {
+                    return method.invoke(orig, args);
+                } finally {
+                    if (startGet != null) {
+                        recordGetTime(startGet);
+                    }
+                }
```

Comment:
Why do we need to use a dynamic proxy instead of just implementing the ConnectionRequest interface?

Reply:
Good point. I don't see any reason we must use a proxy here since this is an interface and not a final class. I was just taking advantage of the existing infrastructure that was seemingly awaiting this metric to be plugged in. :) But let me modify it to just use a simple delegation scheme.

Reply:
Updated to use delegation instead. I think the primary reason to use a proxy is that it would automatically support any new interface methods that are added. I don't know how likely that is to happen with these interfaces.